python/tk_multi_workfiles/step_list_filter.py

```diff
@@ -124,6 +124,11 @@ def select_all_steps(self, value=True):
             # Only toggle widgets currently active
             if not self._list_widget.isRowHidden(item_row):
                 item = self._list_widget.item(item_row)
+                item_step = item.data(QtCore.Qt.UserRole)
```

Comment:
normally we always have a constant here to scope what we are actually storing e..g.

```python
item.data(QtCore.Qt.UserRole+self.PIPELINE_STEP_ID)
```

Either that, or a comment outlining what's being stored would be good.